### ISSUE TITLE
OS-CNN Implementation

### ISSUE THREAD

**Issue by james-large** [sktime/sktime#5828]
Original code is in pytorch

Paper: https://arxiv.org/pdf/2002.10061.pdf

Code: https://github.com/Wensi-Tang/OS-CNN/

**Comment by SABARNO-PRAMANICK** [comment]
hey, has this issue been fixed?

**Comment by fkiraly** [comment]
No, @SABARNO-PRAMANICK, the estimator has not been added yet.

I also note, the linked repository does not have a license, so in-principle it has strong copyright on it. We should contact the original author, @wensi-tang, whether they are willing to release their code under a permissive license, for integration in `sktime`.

**Comment by SABARNO-PRAMANICK** [comment]
@fkiraly , since the issue is connected to licensing, can I contribute to this? 
Should adding the estimator be enough?

**Comment by fkiraly** [comment]
@SABARNO-PRAMANICK, there is existing code here, this looks like a reimplementation: https://github.com/sktime/sktime-dl/pull/111

You could start with the Pr and move it to sktime.

**Comment by SABARNO-PRAMANICK** [comment]
> @SABARNO-PRAMANICK, there is existing code here, this looks like a reimplementation: [sktime/sktime-dl#111](https://github.com/sktime/sktime-dl/pull/111)
> 
> You could start with the Pr and move it to sktime.

Hi @fkiraly , I'm working on it, but just to be sure, do you want me to update the already existing code for `OS-CNN` from `sktime-dl` into the `sktime` repo?

1.  I would need to create multiple `.py` files in the sktime repo.
2. Files changes including new files would be:
![image](https://github.com/user-attachments/assets/650ef52a-deb7-4ba1-bcec-e8560aadc02d)

3. There are other file dependencies required for this pr such as :
                   ` sktime_dl/regression/_regressor.py`
                   `sktime_dl/classification/_classifier`
                   `sktime_dl/networks/_network`

**Comment by fkiraly** [comment]
Yes, @SABARNO-PRAMANICK, the way would be to move the files - however there is already `BaseClassifier` and `BaseRegressor` in `sktime`, you could look at how the other `tensorflow` based estimators are handled.

Regarding the `utils`, what would we need in `sktime`?

**Comment by SABARNO-PRAMANICK** [comment]
> Yes, @SABARNO-PRAMANICK, the way would be to move the files - however there is already `BaseClassifier` and `BaseRegressor` in `sktime`, you could look at how the other `tensorflow` based estimators are handled.
> 
> Regarding the `utils`, what would we need in `sktime`?

Thank you for the instructions @fkiraly , I need help with understanding a few things so I'll comment it here as well as in the discord thread for better comms;
1. As for your instructions, should I utilise the already developed `BaseClassifier` and `BaseRegressor` in the new files for `OS-CNN` instead of the `BaseDeepClassifier ` and `BaseDeepRegressor` ?
2. What should be done about the missing dependency on : `from sktime_dl.networks._network import BaseDeepNetwork` ?
3. Regarding the `utils ` ;
`from sktime_dl.utils import check_and_clean_data`
`from sktime_dl.utils import check_is_fitted`
`from sktime_dl.utils import save_trained_model`

These are the `utils `files required in the utils.
Also, could you please refer me some other `tensorflow `based model that I can take reference from?

**Comment by fkiraly** [comment]
comments are in https://github.com/sktime/sktime/pull/7559, thanks for the PR!
